src/aggregate/aggregate_exec.c

```diff
@@ -1067,23 +1067,24 @@ static void runCursor(RedisModule_Reply *reply, Cursor *cursor, size_t num) {
   RedisSearchCtx_UnlockSpec(req->sctx); // Verify that we release the spec lock
 
   if (req->stateflags & QEXEC_S_ITERDONE) {
-    AREQ_Free(req);
-    cursor->execState = NULL;
     Cursor_Free(cursor);
   } else {
     // Update the idle timeout
     Cursor_Pause(cursor);
   }
 }
 
-static void cursorRead(RedisModule_Reply *reply, uint64_t cid, size_t count) {
+static void cursorRead(RedisModule_Reply *reply, uint64_t cid, size_t count, bool bg) {
   Cursor *cursor = Cursors_TakeForExecution(GetGlobalCursor(cid), cid);
 
   if (cursor == NULL) {
     RedisModule_Reply_Error(reply, "Cursor not found");
     return;
   }
   QueryError status = {0};
+  AREQ *req = cursor->execState;
+  req->qiter.err = &status;
```

Comment:
No, because after returning from `runCursor` in this function (and after returning from this function) the reply to the client should be finalized already.